lib/media_types/scheme/rules.rb

```diff
@@ -111,10 +112,14 @@ def inspect_format_attribute(indent, key, value = self[key])
       private
 
       attr_accessor :allow_empty, :optional_keys
-      attr_writer :expected_type
+      attr_writer :expected_type, :expected_key_type
 
-      def normalize_key(key)
-        String(key).to_sym
+      def verify_type_and_normalize_key(key, doVerify = true)
+        normalized_key = String(key).to_sym
+        if doVerify && !key.instance_of?(expected_key_type)
+          raise KeyTypeError, "Key is of the wrong type. Expected #{expected_key_type} but got #{key.class}" unless __getobj__[normalized_key].instance_of?(NotStrict)
+        end
+        return normalized_key
```

Comment:
```suggestion
        normalized_key
```

In ruby, you don't need to `explicitly` return at the end of a method.

Reply:
Old habits and such. Will update when splitting the method.